File: Free SysLog/Support Code/My Data Classes.vb
Imports Free_SysLog.SupportCode

Public Class SavedData
    Public time, ip, log, fileName, logType, hostname, appName, rawLogData, alertText As String
    Public alertType As AlertType
    Public DateObject, ServerDate As Date
    Public BoolAlerted As Boolean = False

    Public Function MakeDataGridRow(ByRef dataGrid As DataGridView, height As Integer) As MyDataGridViewRow
        Using MyDataGridViewRow As New MyDataGridViewRow
            With MyDataGridViewRow
                .CreateCells(dataGrid)
                .Cells(ColumnIndex_ComputedTime).Value = time
                .Cells(ColumnIndex_ComputedTime).Style.Alignment = DataGridViewContentAlignment.MiddleCenter
                .Cells(ColumnIndex_LogType).Value = If(String.IsNullOrWhiteSpace(logType), "", logType)
                .Cells(ColumnIndex_IPAddress).Value = ip
                .Cells(ColumnIndex_RemoteProcess).Value = If(String.IsNullOrWhiteSpace(appName), "", appName)
                .Cells(ColumnIndex_Hostname).Value = If(String.IsNullOrWhiteSpace(hostname), "", hostname)
                .Cells(ColumnIndex_LogText).Value = log
                .Cells(ColumnIndex_Alerted).Value = If(BoolAlerted, "Yes", "No")
                .Cells(ColumnIndex_Alerted).Style.Alignment = DataGridViewContentAlignment.MiddleCenter
                .Cells(ColumnIndex_Alerted).Style.WrapMode = DataGridViewTriState.True
                .Cells(ColumnIndex_ServerTime).Value = If(ServerDate = Date.MinValue, "", ToIso8601Format(ServerDate))
                .DateObject = DateObject
                .BoolAlerted = BoolAlerted
                .MinimumHeight = height
                .RawLogData = rawLogData
                .AlertText = alertText
                .alertType = alertType

                If My.Settings.font IsNot Nothing Then
                    .Cells(ColumnIndex_ComputedTime).Style.Font = My.Settings.font
                    .Cells(ColumnIndex_LogType).Style.Font = My.Settings.font
                    .Cells(ColumnIndex_IPAddress).Style.Font = My.Settings.font
                    .Cells(ColumnIndex_RemoteProcess).Style.Font = My.Settings.font
                    .Cells(ColumnIndex_Hostname).Style.Font = My.Settings.font
                    .Cells(ColumnIndex_LogText).Style.Font = My.Settings.font
                    .Cells(ColumnIndex_Alerted).Style.Font = My.Settings.font
                    .Cells(ColumnIndex_ServerTime).Style.Font = My.Settings.font
                End If
            End With

            Return MyDataGridViewRow
        End Using
    End Function
End Class

Public Class ReplacementsClass
    Public BoolRegex As Boolean
    Public BoolCaseSensitive As Boolean
    Public StrReplace, StrReplaceWith As String
    Public BoolEnabled As Boolean = True

    Public Function ToListViewItem() As MyReplacementsListViewItem
        Dim listViewItem As New MyReplacementsListViewItem(StrReplace)
        listViewItem.SubItems.Add(StrReplaceWith)
        listViewItem.SubItems.Add(If(BoolRegex, "Yes", "No"))
        listViewItem.SubItems.Add(If(BoolCaseSensitive, "Yes", "No"))
        listViewItem.SubItems.Add(If(BoolEnabled, "Yes", "No"))
        listViewItem.BoolRegex = BoolRegex
        listViewItem.BoolCaseSensitive = BoolCaseSensitive
        listViewItem.BoolEnabled = BoolEnabled
        If My.Settings.font IsNot Nothing Then listViewItem.Font = My.Settings.font
        Return listViewItem
    End Function
End Class

Public Class AlertsHistory
    Public strTime, strAlertText, strIP, strLog, strRawLog As String
    Public alertType As AlertType

    Public Function MakeDataGridRow(ByRef dataGrid As DataGridView, height As Integer) As AlertsHistoryDataGridViewRow
        Using AlertsHistoryDataGridViewRow As New AlertsHistoryDataGridViewRow
            With AlertsHistoryDataGridViewRow
                .CreateCells(dataGrid)
                .Cells(0).Value = strTime
                .Cells(0).Style.Alignment = DataGridViewContentAlignment.MiddleCenter

                If alertType = AlertType.ErrorMsg Then
                    .Cells(1).Value = "Error"
                ElseIf alertType = AlertType.Warning Then
                    .Cells(1).Value = "Warning"
                ElseIf alertType = AlertType.Info Then
                    .Cells(1).Value = "Information"
                Else
                    .Cells(1).Value = ""
                End If

                .Cells(1).Style.Alignment = DataGridViewContentAlignment.MiddleCenter
                .Cells(2).Value = strAlertText

                If My.Settings.font IsNot Nothing Then
                    .Cells(0).Style.Font = My.Settings.font
                    .Cells(1).Style.Font = My.Settings.font
                    .Cells(2).Style.Font = My.Settings.font
                End If

                .strRawLog = strRawLog
                .strIP = strIP
                .strLog = strLog
                .strTime = strTime
                .strAlertText = strAlertText
            End With

            Return AlertsHistoryDataGridViewRow
        End Using
    End Function
End Class

Public Class IgnoredClass
    Public BoolRegex As Boolean
    Public BoolCaseSensitive As Boolean
    Public StrIgnore As String
    Public BoolEnabled As Boolean = True

    Public Function ToListViewItem() As MyIgnoredListViewItem
        Dim listViewItem As New MyIgnoredListViewItem(StrIgnore)
        listViewItem.SubItems.Add(If(BoolRegex, "Yes", "No"))
        listViewItem.SubItems.Add(If(BoolCaseSensitive, "Yes", "No"))
        listViewItem.SubItems.Add(If(BoolEnabled, "Yes", "No"))
        listViewItem.BoolRegex = BoolRegex
        listViewItem.BoolCaseSensitive = BoolCaseSensitive
        listViewItem.BoolEnabled = BoolEnabled
        If My.Settings.font IsNot Nothing Then listViewItem.Font = My.Settings.font
        Return listViewItem
    End Function
End Class

Public Enum AlertType As Byte
    Warning
    Info
    ErrorMsg
    None
End Enum

Public Class AlertsClass
    Public BoolRegex As Boolean
    Public BoolCaseSensitive As Boolean
    Public StrLogText, StrAlertText As String
    Public alertType As AlertType = AlertType.None
    Public BoolEnabled As Boolean = True

    Public Function ToListViewItem() As AlertsListViewItem
        Dim listViewItem As New AlertsListViewItem(StrLogText) With {.StrLogText = StrLogText, .StrAlertText = StrAlertText}
        listViewItem.SubItems.Add(If(String.IsNullOrWhiteSpace(StrAlertText), "(Shows Log Text)", StrAlertText))
        listViewItem.SubItems.Add(If(BoolRegex, "Yes", "No"))
        listViewItem.SubItems.Add(If(BoolCaseSensitive, "Yes", "No"))

        Select Case alertType
            Case AlertType.Warning
                listViewItem.SubItems.Add("Warning")
            Case AlertType.ErrorMsg
                listViewItem.SubItems.Add("Error")
            Case AlertType.Info
                listViewItem.SubItems.Add("Information")
            Case AlertType.None
                listViewItem.SubItems.Add("None")
        End Select

        listViewItem.SubItems.Add(If(BoolEnabled, "Yes", "No"))

        listViewItem.BoolRegex = BoolRegex
        listViewItem.BoolCaseSensitive = BoolCaseSensitive
        listViewItem.AlertType = alertType
        listViewItem.BoolEnabled = BoolEnabled
        If My.Settings.font IsNot Nothing Then listViewItem.Font = My.Settings.font
        Return listViewItem
    End Function
End Class

Public Class ProxiedSysLogData
    Public ip, log As String
End Class

Public Class NotificationDataPacket
    Public logtext, alerttext, logdate, sourceip, rawlogtext As String
End Class

Public Class CustomHostname
    Public ip, deviceName As String

    Public Function ToListViewItem() As ListViewItem
        Dim ListViewItem As New ListViewItem(ip)

        With ListViewItem
            .SubItems.Add(deviceName)
        End With

        If My.Settings.font IsNot Nothing Then ListViewItem.Font = My.Settings.font

        Return ListViewItem
    End Function
End Class

Public Class SysLogProxyServer
    Public ip As String
    Public name As String = Nothing
    Public port As Integer
    Public boolEnabled As Boolean = True

    Public Function ToListViewItem() As ServerListViewItem
        Dim ServerListView As New ServerListViewItem(ip)

        With ServerListView
            .SubItems.Add(port.ToString)
            .SubItems.Add(If(boolEnabled, "Yes", "No"))
            .SubItems.Add(name)
            .StrName = name
            .BoolEnabled = boolEnabled
            If My.Settings.font IsNot Nothing Then ServerListView.Font = My.Settings.font
        End With

        Return ServerListView
    End Function
End Class

File: Free SysLog/Support Code/My Data Classes.cs
using System;
using System.Windows.Forms;

namespace Free_SysLog
{

    public class ColumnOrder
    {
        public string ColumnName;
        public int ColumnIndex;
    }

    public class SavedData
    {
        public string time, ip, log, fileName, logType, hostname, appName, rawLogData, alertText;
        public AlertType alertType = AlertType.None;
        public DateTime DateObject, ServerDate;
        public bool BoolAlerted = false;

        public MyDataGridViewRow MakeDataGridRow(ref DataGridView dataGrid)
        {
            using (var MyDataGridViewRow = new MyDataGridViewRow())
            {
                MyDataGridViewRow.CreateCells(dataGrid);
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_ComputedTime].Value = time;
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_ComputedTime].Style.Alignment = DataGridViewContentAlignment.MiddleCenter;
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_LogType].Value = string.IsNullOrWhiteSpace(logType) ? "" : logType;
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_IPAddress].Value = ip;
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_RemoteProcess].Value = string.IsNullOrWhiteSpace(appName) ? "" : appName;
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_Hostname].Value = string.IsNullOrWhiteSpace(hostname) ? "" : hostname;
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_LogText].Value = log;
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_Alerted].Value = BoolAlerted ? "Yes" : "No";
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_Alerted].Style.Alignment = DataGridViewContentAlignment.MiddleCenter;
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_Alerted].Style.WrapMode = DataGridViewTriState.True;
                MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_ServerTime].Value = ServerDate == DateTime.MinValue ? "" : SupportCode.SupportCode.ToIso8601Format(ServerDate);
                MyDataGridViewRow.DateObject = DateObject;
                MyDataGridViewRow.BoolAlerted = BoolAlerted;
                MyDataGridViewRow.RawLogData = rawLogData;
                MyDataGridViewRow.AlertText = alertText;
                MyDataGridViewRow.alertType = alertType;

                if (My.MySettingsProperty.Settings.font is not null)
                {
                    MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_ComputedTime].Style.Font = My.MySettingsProperty.Settings.font;
                    MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_LogType].Style.Font = My.MySettingsProperty.Settings.font;
                    MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_IPAddress].Style.Font = My.MySettingsProperty.Settings.font;
                    MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_RemoteProcess].Style.Font = My.MySettingsProperty.Settings.font;
                    MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_Hostname].Style.Font = My.MySettingsProperty.Settings.font;
                    MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_LogText].Style.Font = My.MySettingsProperty.Settings.font;
                    MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_Alerted].Style.Font = My.MySettingsProperty.Settings.font;
                    MyDataGridViewRow.Cells[SupportCode.SupportCode.ColumnIndex_ServerTime].Style.Font = My.MySettingsProperty.Settings.font;
                }

                MyDataGridViewRow.DefaultCellStyle.Padding = new Padding(0, 2, 0, 2);

                return MyDataGridViewRow;
            }
        }
    }

    public class ReplacementsClass
    {
        public bool BoolRegex;
        public bool BoolCaseSensitive;
        public string StrReplace, StrReplaceWith;
        public bool BoolEnabled = true;

        public MyReplacementsListViewItem ToListViewItem()
        {
            var listViewItem = new MyReplacementsListViewItem(StrReplace);
            listViewItem.SubItems.Add(StrReplaceWith);
            listViewItem.SubItems.Add(BoolRegex ? "Yes" : "No");
            listViewItem.SubItems.Add(BoolCaseSensitive ? "Yes" : "No");
            listViewItem.SubItems.Add(BoolEnabled ? "Yes" : "No");
            listViewItem.BoolRegex = BoolRegex;
            listViewItem.BoolCaseSensitive = BoolCaseSensitive;
            listViewItem.BoolEnabled = BoolEnabled;
            if (My.MySettingsProperty.Settings.font is not null)
                listViewItem.Font = My.MySettingsProperty.Settings.font;
            return listViewItem;
        }
    }

    public class AlertsHistory
    {
        public string strTime, strAlertText, strIP, strLog, strRawLog;
        public AlertType alertType;

        public AlertsHistoryDataGridViewRow MakeDataGridRow(ref DataGridView dataGrid)
        {
            using (var AlertsHistoryDataGridViewRow = new AlertsHistoryDataGridViewRow())
            {
                AlertsHistoryDataGridViewRow.CreateCells(dataGrid);
                AlertsHistoryDataGridViewRow.Cells[0].Value = strTime;
                AlertsHistoryDataGridViewRow.Cells[0].Style.Alignment = DataGridViewContentAlignment.MiddleCenter;

                if (alertType == AlertType.ErrorMsg)
                {
                    AlertsHistoryDataGridViewRow.Cells[1].Value = "Error";
                }
                else if (alertType == AlertType.Warning)
                {
                    AlertsHistoryDataGridViewRow.Cells[1].Value = "Warning";
                }
                else if (alertType == AlertType.Info)
                {
                    AlertsHistoryDataGridViewRow.Cells[1].Value = "Information";
                }
                else
                {
                    AlertsHistoryDataGridViewRow.Cells[1].Value = "";
                }

                AlertsHistoryDataGridViewRow.Cells[1].Style.Alignment = DataGridViewContentAlignment.MiddleCenter;
                AlertsHistoryDataGridViewRow.Cells[2].Value = strAlertText;

                if (My.MySettingsProperty.Settings.font is not null)
                {
                    AlertsHistoryDataGridViewRow.Cells[0].Style.Font = My.MySettingsProperty.Settings.font;
                    AlertsHistoryDataGridViewRow.Cells[1].Style.Font = My.MySettingsProperty.Settings.font;
                    AlertsHistoryDataGridViewRow.Cells[2].Style.Font = My.MySettingsProperty.Settings.font;
                }

                AlertsHistoryDataGridViewRow.strRawLog = strRawLog;
                AlertsHistoryDataGridViewRow.strIP = strIP;
                AlertsHistoryDataGridViewRow.strLog = strLog;
                AlertsHistoryDataGridViewRow.strTime = strTime;
                AlertsHistoryDataGridViewRow.strAlertText = strAlertText;

                return AlertsHistoryDataGridViewRow;
            }
        }
    }

    public class IgnoredClass
    {
        public bool BoolRegex;
        public bool BoolCaseSensitive;
        public string StrIgnore;
        public bool BoolEnabled = true;

        public MyIgnoredListViewItem ToListViewItem()
        {
            var listViewItem = new MyIgnoredListViewItem(StrIgnore);
            listViewItem.SubItems.Add(BoolRegex ? "Yes" : "No");
            listViewItem.SubItems.Add(BoolCaseSensitive ? "Yes" : "No");
            listViewItem.SubItems.Add(BoolEnabled ? "Yes" : "No");
            listViewItem.BoolRegex = BoolRegex;
            listViewItem.BoolCaseSensitive = BoolCaseSensitive;
            listViewItem.BoolEnabled = BoolEnabled;
            if (My.MySettingsProperty.Settings.font is not null)
                listViewItem.Font = My.MySettingsProperty.Settings.font;
            return listViewItem;
        }
    }

    public enum AlertType : byte
    {
        Warning,
        Info,
        ErrorMsg,
        None
    }

    public class AlertsClass
    {
        public bool BoolRegex;
        public bool BoolCaseSensitive;
        public string StrLogText, StrAlertText;
        public AlertType alertType = AlertType.None;
        public bool BoolEnabled = true;
        public bool BoolLimited = true;

        public AlertsListViewItem ToListViewItem()
        {
            var listViewItem = new AlertsListViewItem(StrLogText) { StrLogText = StrLogText, StrAlertText = StrAlertText };
            listViewItem.SubItems.Add(string.IsNullOrWhiteSpace(StrAlertText) ? "(Shows Log Text)" : StrAlertText);
            listViewItem.SubItems.Add(BoolRegex ? "Yes" : "No");
            listViewItem.SubItems.Add(BoolCaseSensitive ? "Yes" : "No");

            switch (alertType)
            {
                case AlertType.Warning:
                    {
                        listViewItem.SubItems.Add("Warning");
                        break;
                    }
                case AlertType.ErrorMsg:
                    {
                        listViewItem.SubItems.Add("Error");
                        break;
                    }
                case AlertType.Info:
                    {
                        listViewItem.SubItems.Add("Information");
                        break;
                    }
                case AlertType.None:
                    {
                        listViewItem.SubItems.Add("None");
                        break;
                    }
            }

            listViewItem.SubItems.Add(BoolLimited ? "Yes" : "No");
            listViewItem.SubItems.Add(BoolEnabled ? "Yes" : "No");

            listViewItem.BoolRegex = BoolRegex;
            listViewItem.BoolCaseSensitive = BoolCaseSensitive;
            listViewItem.AlertType = alertType;
            listViewItem.BoolEnabled = BoolEnabled;
            listViewItem.BoolLimited = BoolLimited;
            if (My.MySettingsProperty.Settings.font is not null)
                listViewItem.Font = My.MySettingsProperty.Settings.font;
            return listViewItem;
        }
    }

    public class ProxiedSysLogData
    {
        public string ip, log;
    }

    public class NotificationDataPacket
    {
        public string logtext, alerttext, logdate, sourceip, rawlogtext;
    }

    public class CustomHostname
    {
        public string ip, deviceName;

        public ListViewItem ToListViewItem()
        {
            var ListViewItem = new ListViewItem(ip);

            ListViewItem.SubItems.Add(deviceName);

            if (My.MySettingsProperty.Settings.font is not null)
                ListViewItem.Font = My.MySettingsProperty.Settings.font;

            return ListViewItem;
        }
    }

    public class SysLogProxyServer
    {
        public string ip;
        public string name = null;
        public int port;
        public bool boolEnabled = true;

        public ServerListViewItem ToListViewItem()
        {
            var ServerListView = new ServerListViewItem(ip);

            ServerListView.SubItems.Add(port.ToString());
            ServerListView.SubItems.Add(boolEnabled ? "Yes" : "No");
            ServerListView.SubItems.Add(name);
            ServerListView.StrName = name;
            ServerListView.BoolEnabled = boolEnabled;
            if (My.MySettingsProperty.Settings.font is not null)
                ServerListView.Font = My.MySettingsProperty.Settings.font;

            return ServerListView;
        }
    }
}